### ISSUE TITLE
Upgrade the library to support Spring boot 3.x.

### ISSUE THREAD

**Issue by scottf** [nats-io/spring-nats#42]
This will require a major version change.

**Comment by YunaBraska** [comment]
isn't this alreadt done here: https://github.com/nats-io/spring-nats/blob/main/nats-spring/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports

**Comment by kunalvarpe** [comment]
> isn't this alreadt done here: https://github.com/nats-io/spring-nats/blob/main/nats-spring/src/main/resources/META-INF/spring/org.springframework.boot.autoconfigure.AutoConfiguration.imports

You are right but it will not only require this there are few other changes too. Also spring boot 3.x and spring cloud need some compatibility changes.

**Comment by scottf** [comment]
@kunalvarpe please DM me on nats slack

**Comment by maryia-haladko** [comment]
Hi colleagues! Could you tell when this version can be expected in the maven central?

**Comment by zouhuigang** [comment]
When will springboot3.4.1 be supported?

**Comment by scottf** [comment]
@zouhuigang Isn't the library already at 3.5?
```
<spring-boot.version>3.5.0</spring-boot.version>
```
